src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -347,13 +535,23 @@ const executeQueryBase = async (
 
     const dataset = services.data.dataViews.convertToDataset(dataView);
 
+    // Create histogram config once for use in both query building and result processing
+    let histogramConfig: HistogramConfig | null = null;
+    if (isHistogramQuery) {
+      histogramConfig = createHistogramConfigWithInterval(dataView, interval, services, getState);
+    }
+
     const preparedQueryObject = {
       ...query,
       dataset,
-      query: cacheKey,
+      query:
+        isHistogramQuery && histogramConfig
+          ? buildQuery(queryString, histogramConfig, services)
```

Comment:
I think the `createHistogramConfigWithInterval` and `buildQuery` flow might contain a little redundancy here. From a general overview, it seems to me like we only need the timeFieldName from dataView, the interval, and the current query to create the histogram query. Looking at the `createHistogramConfigWithInterval` and `buildQuery` functions, I think they contain a lot of vestigial components from the old DSL way of creating the histogram. 

Could you take another look at these functions and see if you can remove a lot of the unneeded complexity?